platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/reports/processor/IssueReporterProcessor.kt

```diff
@@ -20,19 +22,67 @@ import io.bitdrift.capture.reports.binformat.v1.issue_reporting.Platform
 import io.bitdrift.capture.reports.binformat.v1.issue_reporting.ReportType
 import io.bitdrift.capture.reports.binformat.v1.issue_reporting.SDKInfo
 import io.bitdrift.capture.reports.binformat.v1.issue_reporting.Timestamp
-import io.bitdrift.capture.reports.persistence.IFatalIssueReporterStorage
+import io.bitdrift.capture.reports.persistence.IReporterIssueStorage
 import java.io.InputStream
 import kotlin.time.DurationUnit
 import kotlin.time.toDuration
 
 /**
  * Process reports into a packed format
  */
-internal class FatalIssueReporterProcessor(
-    private val fatalIssueReporterStorage: IFatalIssueReporterStorage,
-    private val clientAttributes: IClientAttributes,
-    private val streamingReportsProcessor: IStreamingReportProcessor,
-) {
+internal object IssueReporterProcessor {
```

Comment:
why is this an object (static)? we should really try to stray away from global objects: it makes it hard to test/mock, they can never be garbage collected, require manual lifecycle tracking (isInitialized), etc

Reply:
yeah let's discuss the alternatives